gordo/machine/dataset/data_provider/ncs_reader.py

```diff
@@ -1,21 +1,99 @@
 # -*- coding: utf-8 -*-
 import logging
+from collections import OrderedDict
 from concurrent.futures import ThreadPoolExecutor
 from datetime import datetime
 from typing import Iterable, List, Optional
 from urllib.parse import quote
 
-import numpy as np
 import pandas as pd
 from azure.datalake.store import core
 
 from gordo.machine.dataset.data_provider.base import GordoBaseDataProvider
+from gordo.machine.dataset.data_provider.file_type import (
+    FileType,
+    CsvFileType,
+    ParquetFileType,
+    TimeSeriesColumns,
+)
+from gordo.machine.dataset.data_provider.azure_utils import is_file
 from gordo.machine.dataset.sensor_tag import SensorTag
 from gordo.util import capture_args
 
 logger = logging.getLogger(__name__)
 
 
+time_series_columns = TimeSeriesColumns("Time", "Value", "Status")
+
+
+class NcsFileLookup:
+    def __init__(self, file_type: FileType):
+        """
+        Creates a files finder.
+
+        Notes
+        -----
+        This implementation related only for :class:`gordo.machine.dataset.data_provider.ndc_reader.NcsReader` not for :class:`gordo.machine.dataset.data_provider.iroc_reader.IrocReader`
+
+        Parameters
+        ----------
+        file_type : FileType
+        """
+        self.file_type = file_type
+
+    def lookup(
+        self, client: core.AzureDLFileSystem, dir_path: str, tag_name: str, year: int
+    ) -> Optional[str]:
+        """
+        Find files with a given file type in Azure Data Lake
+
+        Parameters
+        ----------
+        client: core.AzureDLFileSystem
+            Azure Data Lake client
+        dir_path: str
+            Base directory for finding files
+        tag_name
+            File tag (encoded version)
+        year
+            File year
+        """
+        raise NotImplementedError()
+
+
+class NcsCsvLookup(NcsFileLookup):
+    """
+    Finder for CSV files
+    """
+
+    def __init__(self):
+        header = ["Sensor", "Value", "Time", "Status"]
+        super().__init__(CsvFileType(header, time_series_columns))
+
+    def lookup(
+        self, client: core.AzureDLFileSystem, dir_path: str, tag_name: str, year: int
+    ) -> Optional[str]:
+        file_extension = self.file_type.file_extension
+        path = f"{dir_path}/{tag_name}_{year}{file_extension}"
```

Comment:
It looks like the path is slightly different than what is supposed to be, before your change the file path was
`f"{tag_base_path}/{tag_name_encoded}/{tag_name_encoded}_{year}.csv"` 
Where `tag_name_encoded = quote(tag.name, safe=" ")`

Reply:
For this function, I am passing already encoded `tag_name` https://github.com/equinor/gordo/blob/parquet_file_format/gordo/machine/dataset/data_provider/ncs_reader.py#L331 . I have mentioned this [here](https://github.com/equinor/gordo/blob/parquet_file_format/gordo/machine/dataset/data_provider/ncs_reader.py#L57)